### PR TITLE
Add SMAP to IODA converter

### LINKED ISSUE #614: SMAP

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Adds SMAP Salinity to IODA converter, associated ctest, and ctest data

Also tested on original SMAP data file, with plausible results

Fixes https://github.com/NOAA-EMC/GDASApp/issues/614

### CODE REVIEW

File: utils/obsproc/Smap2Ioda.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+      // Set the int metadata names
+      // TODO(AFE) fill this in: intMetadataNames = {"pass", "cycle", "mission"};
+      std::vector<std::string> intMetadataNames = {};
+
+      // Set the float metadata name
+      // TODO(AFE) fill this in:  floatMetadataNames = {"mdt"};
+      std::vector<std::string> floatMetadataNames = {};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
```
Comment (guillaumevernieres) [inline utils/obsproc/Smap2Ioda.h]: same comment that I made in @apchoiCMD 's PR:
Change L49 to
```c++
gdasapp::IodaVars iodaVars(nobs, {}, {});
```
and remove L41-47. It's easy enough to add back later.

File: utils/obsproc/Smap2Ioda.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      int loc;
+      for (int i = 0; i < dim0; i++) {
+        for (int j = 0; j < dim1; j++) {
+          loc = i * dim1 + j;
+          iodaVars.longitude(loc) = lon[i][j];
+          iodaVars.latitude(loc) = lat[i][j];
+          iodaVars.obsVal(loc) = sss[i][j];
+          iodaVars.obsError(loc) = sss_error[i][j];
+          iodaVars.preQc(loc) = sss_qc[i][j];
+          iodaVars.datetime(loc) =  static_cast<int64_t>(obsTime[j] + unixStartDay);
```
Comment (guillaumevernieres) [inline utils/obsproc/Smap2Ioda.h:90]: I'm uneasy with time manipulation ... Did you check that the output is correct?

File: utils/test/testinput/gdas_smap2ioda.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+provider: SMAP
+window begin: 2018-04-15T06:00:00Z
+window end: 2018-04-15T12:00:00Z
+output file: sss_smap.ioda.nc
+input files:
+- sss_smap_1.nc4
+#- sss_smap_2.nc4
```
Comment (guillaumevernieres) [inline utils/test/testinput/gdas_smap2ioda.yaml]: Un-comment L7, the concatenation should work.

**Review by guillaumevernieres — COMMENTED** [review]
Thanks @AndrewEichmann-NOAA . Just a few minor comments.

File: utils/obsproc/Smap2Ioda.h (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Smap2Ioda.h:90]: Yes - the dates in the output (including from the original datafile) landed within the time specified elsewhere in the file

File: utils/obsproc/Smap2Ioda.h (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Smap2Ioda.h]: done

File: utils/obsproc/Smap2Ioda.h (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Smap2Ioda.h:90]: Specifically, I checked the times in a unix epoch converter

File: utils/test/testinput/gdas_smap2ioda.yaml (hunk 3 above)
Comment (AndrewEichmann-NOAA) [reply utils/test/testinput/gdas_smap2ioda.yaml]: done

**Review by guillaumevernieres — APPROVED** [review]
:tada: 
:taco: 's for @AndrewEichmann-NOAA !

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres carnitas o al pastor, por favor

**Pull request merged**